source/MaterialXTest/ShaderValid.cpp

```diff
@@ -385,209 +430,331 @@ static mx::OslValidatorPtr createOSLValidator(bool& orthographicView, std::ostre
-                geomHandler->setIdentifier(shaderBallFile);
-                validator.setLightHandler(lightHandler);
+                log << ">> " << e.what() << "\n";
+                shader = nullptr;
             }
-            else
+            CHECK(shader != nullptr);
+            if (shader == nullptr)
             {
-                const std::string sphereFile(mx::FilePath::getCurrentPath().asString() + "/documents/TestSuite/Geometry/sphere.obj");
-                geomHandler->setIdentifier(sphereFile);
-                validator.setLightHandler(nullptr);
+                log << ">> Failed to generate shader\n";
+                return;
             }
-        
-            validator.validateCreation(shader);
-            validator.validateInputs();
+            CHECK(shader->getSourceCode(mx::HwShader::PIXEL_STAGE).length() > 0);
+            CHECK(shader->getSourceCode(mx::HwShader::VERTEX_STAGE).length() > 0);
 
-            program->printUniforms(log);
-            program->printAttributes(log);
+            if (outputMtlxDoc)
+            {
+                mx::writeToXmlFile(doc, shaderPath + ".mtlx");
+            }
 
-            validator.validateRender(!isShader);
-            std::string fileName = shaderPath + ".exr";
-            validator.save(fileName);
+            // Validate
+            MaterialX::GlslProgramPtr program = validator.program();
+            bool validated = false;
+            try
+            {
+                mx::GeometryHandlerPtr geomHandler = validator.getGeometryHandler();
 
-            validated = true;
-        }
-        catch (mx::ExceptionShaderValidationError e)
-        {
-            // Dump shader stages on error
-            std::ofstream file;
-            file.open(shaderPath + "_vs.glsl");
-            file << shader->getSourceCode(mx::HwShader::VERTEX_STAGE);
-            file.close();
-            file.open(shaderPath + "_ps.glsl");
-            file << shader->getSourceCode(mx::HwShader::PIXEL_STAGE);
-            file.close();
+                bool isShader = mx::elementRequiresShading(element);
+                if (isShader)
+                {
+                    mx::FilePath geomPath;
+                    if (!testOptions.glslShaderGeom.isEmpty())
+                    {
+                        if (!testOptions.glslShaderGeom.isAbsolute())
+                        {
+                            geomPath = mx::FilePath::getCurrentPath() / mx::FilePath("/documents/TestSuite/Geometry") / testOptions.glslShaderGeom;
+                        }
+                        else
+                        {
+                            geomPath = testOptions.glslShaderGeom;
+                        }
+                    }
+                    else
+                    {
+                        geomPath = mx::FilePath::getCurrentPath().asString() + "/documents/TestSuite/Geometry/shaderball.obj";
```

Comment:
Shouldn't this be like above? Namely mx::FilePath::getCurrentPath() / mx::FilePath("documents/TestSuite/Geometry/shaderball.obj");

Reply:
Can be. I'll make the change.